### PR TITLE
Provide auto building of documentation and publishing to gh-pages

### LINKED ISSUE #4: Provide automated build process for documentation into gh-pages
Leverage a process like https://www.alkaline-ml.com/2018-12-23-automate-gh-builds/ that allows for the building of the documentation on merge to master with a push of the HTML artifacts pushed to `gh-pages` branch.

From there, then GitHub pages can serve our documentation contents.

### PR DESCRIPTION
**Pull request by leifmadsen**
Depends on #3 
Closes #4

### CODE REVIEW

**Comment by leifmadsen** [comment]
So this script seems to work locally, and I can push the data using the same exported environment files locally. Git version is the same... so at this point I'm not sure exactly why this is failing in Travis.

Putting it away for now...

File: build_tools/travis.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+# install deps for Fedora
+echo "--- installing dependencies"
+sudo dnf install findutils git make asciidoctor -y
+
+echo "--- building documentation"
+make clean html
+
+# Checkout our gh-pages branch, remove everything but .git
+echo "--- switching to gh-pages"
```
Comment (pabelanger) [inline build_tools/travis.sh]: add the following to next line:

```
git config remote.origin.fetch "+refs/heads/*:refs/remotes/origin/*"
```

**Comment by leifmadsen** [comment]
[Provide auto building of documentation and publishing to gh-pages](https://trello.com/c/UVgbsDJm/704-provide-auto-building-of-documentation-and-publishing-to-gh-pages)

**Comment by leifmadsen** [comment]
BTW this works now. I tested with local overrides to make sure it'll fire when branch is master (should execute on merge of docs to master).

Docs are built to check for errors but a push into gh-pages should happen on master merge. Can't really test that until we merge into master first time.

Ready for review.

**Comment by JoanneOFlynn2018** [comment]
Hi Leif, Thanks for doing all this, you really didn't have to but I appreciate you getting it set up so quickly. I forked another project so that's why you're seeing references to foreman. The topics directory was in the other repo, not in ours.

File: common/Makefile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-BUILD = foreman
+BUILD = common
```
Comment (JoanneOFlynn2018) [inline common/Makefile]: The team on the foreman project referred to upstream as foreman and downstream as satellite. I was going to simply use upstream and downstream. What do you think? In this case here, is build pointing to an actual directory or is it the name of the condition?

Comment (leifmadsen) [reply common/Makefile]: Actually I think this `Makefile` in the `common/` directory isn't doing anything, and that we can just drop it. I had named the `BUILD = saf` in the `doc-Service-Assurance-Framework/` directory. I'm not sure if there is a reason we should really both with changing between the two?

Comment (JoanneOFlynn2018) [reply common/Makefile]: We will need upstream and downstream for the attributes. That is if you want to use upstream terminology upstream and RH terminology downstream.

Comment (leifmadsen) [reply common/Makefile]: Ahhh that makes sense. Looked at the `attributes.adoc` again and now i get it. Changed!

**Pull request merged**